// @toolpad/core/node/shared/navigation.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getItemTitle = exports.getItemKind = void 0;
exports.getPageItemFullPath = getPageItemFullPath;
exports.hasSelectedNavigationChildren = hasSelectedNavigationChildren;
exports.isPageItem = void 0;
exports.isPageItemSelected = isPageItemSelected;
var _pathToRegexp = require("path-to-regexp");
const getItemKind = item => {
  var _item$kind;
  return (_item$kind = item.kind) != null ? _item$kind : 'page';
};
exports.getItemKind = getItemKind;
const isPageItem = item => getItemKind(item) === 'page';
exports.isPageItem = isPageItem;
const getItemTitle = item => {
  var _ref, _item$title;
  return isPageItem(item) ? (_ref = (_item$title = item.title) != null ? _item$title : item.segment) != null ? _ref : '' : item.title;
};
exports.getItemTitle = getItemTitle;
function getPageItemFullPath(basePath, navigationItem) {
  var _navigationItem$segme;
  return `${basePath}${basePath && !navigationItem.segment ? '' : '/'}${(_navigationItem$segme = navigationItem.segment) != null ? _navigationItem$segme : ''}`;
}
function isPageItemSelected(navigationItem, basePath, pathname) {
  return navigationItem.pattern ? (0, _pathToRegexp.pathToRegexp)(`${basePath}${navigationItem.pattern}`).test(pathname) : getPageItemFullPath(basePath, navigationItem) === pathname;
}
function hasSelectedNavigationChildren(navigationItem, basePath, pathname) {
  if (isPageItem(navigationItem) && navigationItem.children) {
    const navigationItemFullPath = getPageItemFullPath(basePath, navigationItem);
    return navigationItem.children.some(nestedNavigationItem => {
      if (!isPageItem(nestedNavigationItem)) {
        return false;
      }
      if (nestedNavigationItem.children) {
        return hasSelectedNavigationChildren(nestedNavigationItem, navigationItemFullPath, pathname);
      }
      return isPageItemSelected(nestedNavigationItem, navigationItemFullPath, pathname);
    });
  }
  return false;
}
